src/mantine-core/src/TransferList/RenderList/RenderList.tsx

```diff
@@ -36,6 +36,7 @@ export interface RenderListProps extends DefaultProps<RenderListStylesNames> {
   limit?: number;
   transferIcon?: React.FunctionComponent<{ reversed }>;
   transferAllIcon?: React.FunctionComponent<{ reversed }>;
+  transferAllMatchingFilter: boolean;
```

Comment:
I think `transferAllMatchingSearch` might make more sense to users, but I named it `Filter` for now since that it what the fitlering prop is called. Let me know if you have any preference.